database/src/ops/block.rs

```diff
@@ -1,89 +1,555 @@
+            }
+        } // for each input
+
+        // Remove each output in the transaction.
+        for (i, output) in tx.prefix.outputs.into_iter().enumerate() {
+            // Outputs with clear amounts.
+            if let Some(amount) = output.amount {
+                // RingCT miner outputs.
+                if miner_tx && tx.prefix.version == 2 {
+                    let amount_index = get_rct_num_outputs(tables.rct_outputs())? - 1;
+                    remove_rct_output(&amount_index, tables.rct_outputs_mut())?;
+                // Pre-RingCT outputs.
+                } else {
+                    let amount_index = tables.num_outputs_mut().take(&amount)?;
+                    remove_output(
+                        &PreRctOutputId {
+                            amount,
+                            amount_index,
+                        },
+                        tables,
+                    )?;
+                }
+            // RingCT outputs.
+            } else {
+                let amount_index = get_rct_num_outputs(tables.rct_outputs())? - 1;
+                remove_rct_output(&amount_index, tables.rct_outputs_mut())?;
+            }
+        } // for each output
+    } // for each transaction
+
+    Ok((block_height, block_hash))
 }
 
-/// TODO
-pub fn get_block() {
-    todo!()
+//---------------------------------------------------------------------------------------------------- `get_block_extended_header_*`
+/// Retrieve a [`ExtendedBlockHeader`] from the database.
+///
+/// This extracts all the data from the database tables
+/// needed to create a full `ExtendedBlockHeader`.
+#[doc = doc_error!()]
+#[inline]
+pub fn get_block_extended_header(
+    block_hash: &BlockHash,
+    tables: &impl Tables,
+) -> Result<ExtendedBlockHeader, RuntimeError> {
+    let height = tables.block_heights().get(block_hash)?;
+    let block_info = tables.block_infos().get(&height)?;
+    let block_blob = tables.block_blobs().get(&height)?.0;
+    let block = Block::read(&mut block_blob.as_slice())?;
+
+    // INVARIANT: #[cfg] @ lib.rs asserts `usize == u64`
+    #[allow(clippy::cast_possible_truncation)]
+    Ok(ExtendedBlockHeader {
+        version: block.header.major_version,
+        vote: block.header.minor_version,
+        timestamp: block.header.timestamp,
+        cumulative_difficulty: block_info.cumulative_difficulty,
+        block_weight: block_info.weight as usize,
+        long_term_weight: block_info.long_term_weight as usize,
+    })
 }
 
-/// TODO
-pub fn get_block_from_height() {
-    todo!()
+/// Same as [`get_block_extended_header`] but with a [`BlockHeight`].
+///
+/// Note: This is more expensive than the above.
+#[doc = doc_error!()]
+#[inline]
+pub fn get_block_extended_header_from_height(
+    block_height: &BlockHeight,
+    tables: &impl Tables,
+) -> Result<ExtendedBlockHeader, RuntimeError> {
+    get_block_extended_header(&tables.block_infos().get(block_height)?.block_hash, tables)
 }
```

Comment:
These functions should be the other way around IMO `get_block_extended_header` should call `get_block_extended_header_from_height` after getting the blocks height from hash